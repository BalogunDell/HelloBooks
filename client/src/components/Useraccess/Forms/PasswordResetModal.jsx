import React from 'react';
import { connect } from 'react-redux';
import Loader from '../../presentational/Loader';
import { sendEmail } from '../../../Actions/userAccessAction';

/**
 * @description PasswordResetModal
 * 
 * @class PasswordResetModal
 * 
 * @extends {Component}
 */
export class PasswordResetModal extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      loader:false,
      disableBtn: false
    }

    this.handleInput = this.handleInput.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.cancelPassReset = this.cancelPassReset.bind(this);
    const message = document.getElementById('message');
  }

  /**
   * @description User input handler - handleInput
   * 
   * @param { object } event 
   * 
   * @memberof PasswordResetModal
   * 
   * @returns { object } a new state with use email
   */
  handleInput(event) {
    this.setState({email: event.target.value})
  }

  /**
   * @description Reset password handler - handleSubmit
   * 
   * @param { object } event 
   * 
   * @memberof PasswordResetModal
   * 
   * @returns { object } updated state and success message 
   */
  handleSubmit(event) {
     event.preventDefault();
     message.innerHTML = ``;
    this.setState({  disableBtn: true});
    if(this.state.email === "") {
      message.innerHTML = `Your email is needed to reset your password`;
      message.className="red-text center-align";
      this.setState({ disableBtn: false});
    } else {
      
      this.props.sendEmail({email: this.state.email})
      .then(() => {
        Materialize.toast(
          `A password reset link has been sent to your email`,
        3200,
        'blue rounded');

        setTimeout(() => {
          $('.modal').modal('close');
          this.setState({
            disableBtn: true,
            email: ''
          });
        }, 2500);

      })
      .catch((error) => {
        this.setState({ disableBtn: false});
      })
    }
  }

  /**
   * @description Cancel Password Reset - cancelPassReset
   * 
   * @memberof PasswordResetModal
   * 
   * @returns { object } updated state and success message 
   */
  cancelPassReset() {
    this.setState({email: '', loader: false, disableBtn: false
  });
    message.innerHTML = 'A password reset link will be sent to this email';
    message.className="black-text center-align"
  }

  /**
   * @description Lifecycle hook - componentDidMount
   * 
   * @memberof PasswordResetModal
   * 
   * @returns { object } updated state
   */
  componentDidMount() {
    message.innerHTML = '';
    this.setState({
      disableBtn: false,
      error: false
    });
    $(document).ready(() => {
      $('.modal').modal({
        dismissible: false,
        opacity: 0.3
      });
    });
  }

  /**
   * @description React render method - render
   * 
   * @memberof PasswordResetModal
   * 
   * @returns { object } JSX representation of DOM
   */
  render() {
    return(
    <div className="modal passwordModal" id="passwordResetModal">
      <div className="container">
        <div className="col s12 m6 l4 offset-l8">
          <div className="modal-content">
            <h5 className="center-align">Password reset</h5>
              <p className="center-align" id="message">
              A password reset link will be sent to this email</p>
              {
                this.state.error 
                ? 
                  <div className="row center-align">
                    <Loader/>
                  </div>
                :
                null
              }
            <form
              className="col s6 offset-s3"
              id ="handleSubmit"
              onSubmit={this.handleSubmit}>
              <div className="input-field">
                <label htmlFor="email">Email</label>
                <input 
                  type="email"
                  className="validate"
                  value={this.state.email}
                  onChange={this.handleInput}/>
              </div>

              <div className="input-field">
                <button
                  className="btn waves-effect waves-ripple green"
                  disabled= {this.state.disableBtn}>Reset
                </button>
                <div
                  className="btn waves-effect modal-close modal-action"
                  id="cancelPass"
                  onClick={this.cancelPassReset}>Cancel
                </div>
              </div>
            </form>
          </div>   
        </div>
      </div>
    </div>
    );
  }

}

/**
 * @description Redux connect parameter - dispatchToProps
 * 
 * @param {function} dispatch
 * 
 * @returns {object} mapped dispatch 
*/
export const dispatchToProps = (dispatch) => {
  return {
    sendEmail: (email) => dispatch(sendEmail(email))
  }
}

export default connect(null, dispatchToProps)(PasswordResetModal);