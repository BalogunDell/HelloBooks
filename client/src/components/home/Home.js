import React from 'react';
import Background from '../Background/Background';
import Navbar from '../navbar/Navbar';
import { Link } from 'react-router-dom';

/**
 * @class Home
 * @classdesc Returns the Home component
 */
export default class Home extends React.Component {
  render() {
    return(
    <div>
        <div className="container">
            <div className="intro">
              {/*  This holds the intro message and the title  */}
              <h1>Hellobooks Library</h1>
              <p>Hello books allows you to borrow books that have been gathered from 
              different part of the  world. Finding books of your choice 
            just got easier with us</p>
              <Link className=" button btn waves-effect waves-ripple" to ="/about">Get to know more</Link>
              <Link className="button btn waves-effect waves-ripple" to ="/register">Signup</Link>
          </div>
          <div className="arrow-down center">
            <a href="#">
              <i className="fa fa-arrow-circle-down fa-4x arrow"></i>
            </a>
          </div>
        </div>
    </div>
    );
  }
}