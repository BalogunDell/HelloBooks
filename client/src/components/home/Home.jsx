import React from 'react';
import Background from '../Background/Background';
import Navbar from '../navbar/Navbar';
import { Link } from 'react-router-dom';

/**
 * @class Home
 * @classdesc Returns the Home component
 */
export default class Home extends React.Component {
  render() {
    return(
    <div>
        <div className="container">
            <div className="intro">
              {/*  This holds the intro message and the title  */}
              <h1>Hellobooks Library</h1>
              <p>Hello books allows you to borrow books that have been gathered from 
              different part of the  world. Finding books of your choice 
            just got easier with us</p>
              <div className ="row">
                <div className= "col s12 m12 l6">
                  <Link className="btn waves-effect waves-ripple" to ="/about">Get to know more</Link>  
                </div>
                <div className= "col s12 m12 l6">
                  <Link className="btn waves-effect waves-ripple" to ="/register">Signup</Link>
                </div>
              </div>
          </div>
        </div>
    </div>
    );
  }
}