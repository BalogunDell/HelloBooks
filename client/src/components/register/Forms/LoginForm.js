import React from 'react';

/**
 * @class LoginForm
 * @classdesc returns login form
 */
class LoginForm extends React.Component {
  render() {
    return(
<div className="col row">
  <div className="col s12 m6 offset-m3">
    <div className="form-holder">

    {/* form headers  */}
      <div className="form-header">
        <h2>Signin</h2>
        <div>
          <p>Not registered? <a href="#">Signup</a></p>
        </div>
      </div>

    {/* form header ends  */}
      <div>
        <form>
          {/* Username input  */}
          <div className="row">
            <div className="input-field s12">
              <label>Username <span>*</span></label>
              <input type="text" id="username" ref="username"/>
            </div>
          </div>
          

          {/* Password input  */}
          <div className="row">
            <div className="input-field s12">
              <label htmlFor="Password">Password<span>*</span></label>
              <input type="password" id="password" name="password"/>
            </div>  
          </div>
          
          {/* Signin button  */}
          <div className="row">
            <div>
              <button className="btn waves-effect waves-teal" id="regBtn">Signin</button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
 </div>
    );
  }

}

export default LoginForm;