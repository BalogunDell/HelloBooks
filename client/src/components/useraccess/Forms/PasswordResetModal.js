import React from 'react';

/**
 * @class PasswordResetModal
 * @classdesc returns Password reset modal
 */
class PasswordResetModal extends React.Component {
  constructor(prop) {
    super();
    this.state = {
      email: ''
    }

    this.handleInput = this.handleInput.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    const message = document.getElementById('message');
  }

  handleInput(event) {
    this.setState({email: event.target.value})
  }

  handleSubmit(event) {
    // validate input
    if(this.state.email === "") {
    message.innerHTML = `Your email is needed to reset your password`;
    message.className="red-text center-align";  
    } else {
    message.innerHTML = `A password reset link has been sent to ${this.state.email}`;
    message.className="green-text center-align";
    }
    // prevent button default action
    event.preventDefault();
  }
  render() {
    return(
    <div className="passwordModal">
      <div className="modal" id="modal-link">
        <div className="modal-content">
          <h5 className="center-align"> Password reset</h5>
            <p className="center-align" id="message">
            A password reset link will be sent to this email</p>
          <form className="col s6 offset-s3" onSubmit={this.handleSubmit}>
            <div className="input-field">
              <label htmlFor="email">Email</label>
              <input type="email" className="validate" value={this.state.email} onChange={this.handleInput}/>
            </div>

            <div className="input-field">
              <input type="submit" className="btn waves-effect waves-ripple green" value="RESET"/>
              <div className="btn waves-effect modal-close modal-action">Cancel</div>
            </div>
          </form>
        </div> 
      </div>
    </div>
    );
  }

}

export default PasswordResetModal;