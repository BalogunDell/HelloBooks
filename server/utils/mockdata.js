const mockData = {

  // Valid user data
  user1: {
    firstname: 'janet',
    lastname: 'teddy',
    email: 'jane5@mail.com',
    username: 'abbey33',
    password: 'abbey2',
    role: 'user',
    membership: 'silver',
    image: 'image2'
  },

  // Valid user data
  userEdit: {
    firstname: 'Biodun',
    lastname: 'Olaoluwa',
    username: 'abbey33333'
  },

  borrowBookData: {
    bookid: 2
  },

  adminData: {
    firstname: 'uadminFirstName',
    lastname: 'adminLastName',
    username: 'userUsername123',
    email: 'adminEmail@gmail.com',
    password: 'adminpassword',
    role: 'admin',
    membership: 'bronze',
    image: 'image1'
  },

  adminSigninData: {
    username: 'adminUsername22',
    password: 'password',
  },

  invaliduser1Login: {
    email: 'user1Email@gmail.com',
    password: 'user1password',
  },

  user1Login: {
    username: 'abbey33',
    password: 'abbey2',
  },

  user1ConflictData: {
    firstname: 'user1FirstName',
    lastname: 'user1LastName',
    email: 'user1Email@gmail.com',
    password: 'user1password',
    role: 'user',
    membership: 'bronze'
  },

  user2: {
    firstname: 'user2FirstName',
    lastname: 'user2LastName',
    username: 'userUsername123',
    email: 'user2Email@gmail.com',
    password: 'user2password',
    role: 'user',
    membership: 'silver'
  },

  user3: {
    firstname: 'user3FirstName',
    lastname: 'user3LastName',
    email: 'user3Email@gmail.com',
    username: 'user1Username123',
    password: 'user3password',
    role: 'user',
    membership: 'gold'
  },

  adminLogin: {
    username: 'user1Username123',
    password: 'adminpassword'
  },


  // Inalid user data
  invalidUser1: {
    firstname: '',
    lastname: 'invaliduser1LastName',
    email: 'invaliduser1Email@gmail.com',
    password: 'user1password',
    role: 'user',
    membership: 'bronze'
  },

  invaliduser2: {
    firstname: 'invaliduser2FirstName',
    lastname: 'invaliduser2LastName',
    email: 'invaliduser2Email@gmail.com',
    password: 'invaliduser2password',
    role: 'user',
    membership: 'silver'
  },

  invaliduser3: {
    firstname: 'invaliduser3FirstName',
    lastname: 'invaliduser3LastName',
    email: 'invalidUser3Email@gmail.com',
    password: 'invaliduser3password',
    role: 'user',
    membership: 'gold'
  },

  invalidadmin: {
    email: 'invalidadminEmailgmail.com',
    password: 'invalidAdminpassword'
  },

  invaliduser: {
    username: '',
    password: ''
  },

  emptyCredence: {
    username: '',
    password: ''
  },

  bookdata: {
    isbn: '1111',
    title: 'React for Beginners',
    author: 'Nelson Brook',
    pages: 1080,
    year: 2010,
    description: 'the books does this and that',
    quantity: 3,
    categoryid: 2,
    visibility: true,
    image: 'https://res.cloudinary.com/djvjxp2am/image/upload/v1507295978/book8_yy9efp.jpg',
    pdf: 'https://res.cloudinary.com/djvjxp2am/image/upload/v1507295978/book8_yy9efp.jpg',
  },

  editBookdata: {
    id: 2,
    isbn: '4392',
    title: 'React for Beginners',
    author: 'Nelson Brook',
    pages: 100,
    year: 2000,
    description: 'the books does this and that',
    quantity: 300,
    categoryid: 5,
    visibility: true,
    image: 'https://res.cloudinary.com/djvjxp2am/image/upload/v1507295978/book8_yy9efp.jpg',
    pdf: 'https://res.cloudinary.com/djvjxp2am/image/upload/v1507295978/book8_yy9efp.jpg',
  },

  invalidBookdata: {
    isbn: '7764',
    title: '',
    author: 'Nelson Brook',
    pages: 123,
    year: 1977,
    description: 'the books does this and that',
    quantity: 32,
    category: 'Self Growth',
    image: 'images/andela.jpg'
  },

  invalidBookdata2: {
    isbn: '#7764',
    title: 'Abbey owns',
    author: 'Nelson Brook',
    pages: '',
    year: '',
    description: 'the books does this and that',
    quantity: 32,
    category: 'Self Growth',
    image: 'images/andela.jpg'
  },

  invalidBookdata3: {
    isbn: '11164',
    title: 'Abbey owns',
    author: 'Nelson Brook',
    pages: 123,
    year: 1768,
    description: '',
    quantity: '',
    category: 'Self Growth',
    image: 'images/andela.jpg'
  },

  invalidBookdata4: {
    isbn: '11164',
    title: 'Abbey owns',
    author: 'Nelson Brook',
    pages: 123,
    year: 1768,
    description: 'The description of the book',
    quantity: 100,
    category: 'Self Growth',
    image: 'images/andela.jpg'
  },

  borrowBook: {
    userid: 1,
    bookid: 1
  },

  failborrowBook: {
    userid: '',
    bookid: ''
  },

  modifyBookData: {
    bookid: 1,
    title: 'Challenges of a bootcamper',
    author: 'Donald Lawrence',
    pages: 453,
    year: 2000,
    description: 'Talks about Andela Bootcamp',
    quantity: 32,
    category: 'Self Growth',
    image: 'images/andela.jpg'
  },

  userID: {
    userid: 1
  }

};

export default mockData;
