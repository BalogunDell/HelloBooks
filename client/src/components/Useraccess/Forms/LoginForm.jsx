import React from 'react';
import { Link } from 'react-router-dom';
import 'jquery';
import GoogleLogin from 'react-google-login';
import PasswordResetModal from './PasswordResetModal';
import Loader from '../../Userprofile/AdminSubComponents/Loader';

require('dotenv').config();

/**
 * renders login form
 *
 * @param {object} userData
 * @param {Function} handleLoginInput
 * @param {Function} loginHandler
 * @param {Function} googleLoginHandler
 * 
 * @returns {JSX}  JSX representation of commponent
 */
const LoginForm = ({ 
  userData,
  handleLoginInput,
  loginHandler,
  googleLoginHandler,
  loginError,
  isLoading }) => {

    // loader
    const isLoadingText = <h5>Loading...</h5>

    // error display
    const errorDisplay = <div className="center red-text">{loginError}</div>

    // check condition
    let showIsLoading = isLoading ?  <Loader/> : ''; 

    // set error status to empty if loader is displayed
    let errorStatus = isLoading ? '' : errorDisplay;

    
    return(
        <div className="user-login-form">
          <div className="col s12 m6 offset-m3 valing-wrapper">
            <div className="form-holder">
              <div className="form-header">
                <h2>Signin</h2>
                <div>
                  <p>Not registered? <Link to="/register">Signup</Link></p>
                </div>
              </div>
              <div className="user-form">
                <form className="" onSubmit={loginHandler}>

                  <div className="row">
                    <div className="input-field s12">
                      <input 
                      type="text" 
                      id="username" 
                      name="username"
                      value = {userData.username}
                      onChange={handleLoginInput}
                      className="validate"
                      required
                      />
                      <label 
                      data-error={loginError}
                      data-success="">Username <span>*</span></label>
                    </div>
                  </div>
                  <div className="row">
                    <div className="input-field s12">
                      <label>Password<span>*</span></label>
                      <input 
                      type="password" 
                      id="password" 
                      name="password"
                      value = {userData.password}
                      onChange = {handleLoginInput}
                      required
                      
                      />
                      <div className="center red-text">
                        {errorStatus}
                        {showIsLoading}
                      </div>
                    </div>  
                  </div>

                  <div className="row resetPassBtn">
                    <a href="#passwordResetModal" 
                    className="button forgotPass modal-trigger">Forgot password
                    </a>
                  </div>
                  <div className="row">
                    <div>
                      <input 
                        type="submit"
                        className="btn waves-effect waves-teal"
                        id="regBtn"
                        value="Signin"/>
                    </div>
                  </div>
                </form>
                <h5><span>or</span></h5>
                   <div className="row">
                    <GoogleLogin
                      clientId = {process.env.GOOGLE_CLIENT_ID}
                      buttonText = 'Signin with Google'
                      onSuccess = {googleLoginHandler}
                      onFailure = {googleLoginHandler}
                      className = 'btn waves-effect waves-teal red'
                    />
                  </div>
              </div>
            </div>
          </div>

  {<PasswordResetModal/>}
 </div>
    );
  }

LoginForm.propTypes = {
 handleLoginInput: React.PropTypes.func.isRequired,
 googleLoginHandler: React.PropTypes.func.isRequired,
 userData: React.PropTypes.object.isRequired,
 loginError: React.PropTypes.string.isRequired,
 isLoading: React.PropTypes.bool.isRequired
}

export default LoginForm;