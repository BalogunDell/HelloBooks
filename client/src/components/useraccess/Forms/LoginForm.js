import React from 'react';
import { Link } from 'react-router-dom';
import 'jquery';
import PasswordResetModal from './PasswordResetModal.jsx';


const LoginForm = ({ userData, handleLoginInput, loginHandler, error, isLoading }) => {

    // loader
    const isLoadingText = <h5>Loading...</h5>

    // error display
    const errorDisplay = <div className="center red-text">{error}</div>

    // check condition
    let showIsLoading = isLoading ?  isLoadingText : ''; 

    // set error status to empty if loader is displayed
    let errorStatus = isLoading ? '' : errorDisplay;

    
    return(
        <div className="user-login-form">
          <div className="col s12 m6 offset-m3 valing-wrapper">
            <div className="form-holder">

            {/* form headers  */}
              <div className="form-header">
                <h2>Signin</h2>
                <div>
                  <p>Not registered? <Link to="/register">Signup</Link></p>
                </div>
              </div>
            {/* form header ends  */}
              <div className="user-form">
                <form className="" onSubmit={loginHandler}>
                  {/* Username input  */}
                  <div className="row">
                    <div className="input-field s12">
                      <input 
                      type="text" 
                      id="username" 
                      name="username"
                      value = {userData.username}
                      onChange={handleLoginInput}
                      className="validate"
                      />
                      <label data-error={error} data-success="">Username <span>*</span></label>
                    </div>
                  </div>
                  

                  {/* Password input  */}
                  <div className="row">
                    <div className="input-field s12">
                      <label>Password<span>*</span></label>
                      <input 
                      type="password" 
                      id="password" 
                      name="password"
                      value = {userData.password}
                      onChange = {handleLoginInput}
                      
                      />
                      <div className="center red-text">
                        {errorStatus}
                        {showIsLoading}
                        </div>
                    </div>  
                  </div>

                  <div className="row resetPassBtn"><a href="#passwordResetModal" 
                  className="button forgotPass modal-trigger">Forgot password
                  </a></div>
                  
                  {/* Signin button  */}
                  <div className="row">
                    <div>
                      <input type="submit" className="btn waves-effect waves-teal" id="regBtn" value="Signin"/>
                    </div>
                  </div>
                </form>
                <h5><span>or</span></h5>
                   <div className="row">
                    <button className="btn waves-effect waves-teal red">Signup with Google+ </button>
                  </div>
              </div>
            </div>
          </div>

  {<PasswordResetModal/>}
 </div>
    );
  }

LoginForm.propTypes = {
 handleLoginInput: React.PropTypes.func.isRequired,
 userData: React.PropTypes.object.isRequired,
 error: React.PropTypes.string.isRequired,
 isLoading: React.PropTypes.bool.isRequired
}

export default LoginForm;