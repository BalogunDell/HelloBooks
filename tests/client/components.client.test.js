import React from 'react';
import  Adapter from 'enzyme-adapter-react-15';
import chai from 'chai';
import renderer from 'react-test-renderer';
import { shallow, mount, render , configure} from 'enzyme';
import jestSnapshot from 'chai-jest-snapshot';

import Home from '../../client/src/components/home/Home';
import LoginForm from '../../client/src/components/useraccess/Forms/LoginForm';
import { PasswordResetModal } from '../../client/src/components/useraccess/Forms/PasswordResetModal';
import * as mockData from './mockdata';

const expect = chai.expect;
chai.use(jestSnapshot);
configure({ adapter: new Adapter() });
jest.mock('react-router-dom');


// ************************ //
// *****COMPONENTS TEST**** //
// ************************ //
describe('component: Home', () => {
    it('renders Home component', () => {
        const wrapper = render(<Home />);
        const testRender = renderer.create(<Home/>);
      expect(testRender.toJSON()).to.matchSnapshot();
      expect(wrapper.find('.intro')).to.have.length(1);
      expect(wrapper.find('h1').text()).to.equal('Hellobooks Library');
      expect(wrapper.find('p').text()).to.equal(mockData.homeParagraph);
    });

  // it('renders Login Form', () => {
  //   const minProps = {
  //     userData: {},
  //     handleLoginInput: jest.fn(),
  //     loginHandler: jest.fn(),
  //     error: false,
  //     isLoading: false
  //   }
  //   const wrapper = render(<LoginForm {...minProps}/>);
  //   expect(wrapper.find('.user-login-form')).to.have.length(1);
  // });

  it('renders Password Reset Modal', () => {
    
    const testRender = renderer.create(<PasswordResetModal/>);
    const wrapper = mount(<PasswordResetModal handleSubmit = {mockFunction} />);
    const mockFunction = wrapper.instance().handleSubmit;
    wrapper.find('button').simulate('click');
    // expect(mockFunction).to.have.property('callCount', 1);
    expect(testRender.toJSON()).to.matchSnapshot();
    expect(wrapper.find(PasswordResetModal)).to.have.length(1);
    expect(wrapper.find('.modal')).to.have.length(1);
    expect(wrapper.contains(<h5 className="center-align">Password reset</h5>)).to.equal(true);

  })
});