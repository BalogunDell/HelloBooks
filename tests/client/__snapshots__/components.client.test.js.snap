// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: Home renders Home component 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="intro"
    >
      <h1>
        Hellobooks Library
      </h1>
      <p>
        Hello books allows you to borrow books that have been gathered from different part of the  world. Finding books of your choice just got easier with us
      </p>
      <div
        className="row"
      >
        <div
          className="col s12 m12 l6"
        />
        <div
          className="col s12 m12 l6"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`component: Home renders Password Reset Modal 1`] = `
<div
  className="modal passwordModal"
  id="passwordResetModal"
>
  <div
    className="container"
  >
    <div
      className="col s12 m6 l4 offset-l8"
    >
      <div
        className="modal-content"
      >
        <h5
          className="center-align"
        >
          Password reset
        </h5>
        <p
          className="center-align"
          id="message"
        >
          A password reset link will be sent to this email
        </p>
        <form
          className="col s6 offset-s3"
          onSubmit={[Function]}
        >
          <div
            className="input-field"
          >
            <label
              htmlFor="email"
            >
              Email
            </label>
            <input
              className="validate"
              onChange={[Function]}
              type="email"
              value=""
            />
          </div>
          <div
            className="input-field"
          >
            <button
              className="btn waves-effect waves-ripple green"
              disabled={false}
            >
              Reset
            </button>
            <div
              className="btn waves-effect modal-close modal-action"
              onClick={[Function]}
            >
              Cancel
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
